### PR TITLE
RouteCRD design spec

### PR DESCRIPTION
**Pull request by stevesloka**
Initial RouteCRD design doc spec. 

Signed-off-by: Steve Sloka <steves@heptio.com>

### CODE REVIEW

File: design/routecrd.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+## Goals
+
+- Enable ingress specification via Kubernetes Custom Resource Definition
+- Provide field level validation on CRD
+
+## Terminology / Spec
+
+Following is an layout of the spec which the CRD will be implemented:
+
+- **apiVersion**: contour.heptio.com/v1alpha
```
Comment (alexbrand) [inline design/routecrd.md]: Should this be `v1alpha1`?

File: design/routecrd.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+- **kind**: Route
+- **metadata**:
+  - **namespace**: Namespace deployed into
+  - **name**: Name of object
+- **status**: 
+  - **currentStatus**: Current Status of the route (e.g. Invalid, Error, etc)
+  - **lastProcessTime**: Time the object was processed by Contour and made available to cluster
+  - **[]errors**: List of errors found in configuration
+- **spec**:
+  - **strategy (Optional)**: Load balancer algorithm type: Round Robin / Weighted Least Request / Random (Note: A strategy defined here will be the default for all routes below unless overridden in route spec)
```
Comment (alexbrand) [inline design/routecrd.md]: Seems like the strategy should be a required field. Unless we have a default strategy that would apply if the strategy is not specified here nor in the routes.

File: design/routecrd.md (hunk 1 above)
Comment (stevesloka) [reply design/routecrd.md]: yup I fixed up in latest

File: design/routecrd.md (hunk 2 above)
Comment (stevesloka) [reply design/routecrd.md]: I added a note that it defaults to `RoundRobin` if not specified.

**Comment by alexbrand** [comment]
LGTM

**Pull request merged**